### ISSUE TITLE
Calendar: Cannot read property 'left' of undefined error if input not visible

### ISSUE THREAD

**Issue by palianah** [primefaces/primefaces#6204]
**Describe the defect**
We use Calendar a little bit different so the Calendar is on DOM Ready set to display:none; And on a Button Click, the Calendar is sliding in from the side and displayed. The Problem is, the overwritten $.datepicker._attachHandlers from primefaces (in jquery-plugins.js) calls initially the this.updateDatePickerPosition() function, which has not a condition to check if input is visible, it expects the input is visible and calls the $.datepicker._findPos function.
In this step we get the: Cannot read property 'left' of undefined error. 

Simple fix could be to check if input is not visible, to return directly in the updateDatePickerPosition method. 
It would be the line 11449 in the jquery-plugins.js:
<code>
this.updateDatePickerPosition = function (g) {
if (!g.input[0].clientWidth) return;
....
}
</code>
With this line the error is gone. 

**Environment:**
 - PF Version: 8.0.3
 - JSF + version: MyFaces 2.2.20
 - Affected browsers: ALL

**Expected behavior**
No error on datepicker input field, even if datepicker input is not visible on DOM Ready.

**Comment by melloware** [comment]
Could you post a little XHTML example your calendar or a reproducer so it can help me see the issue?  I think the fix is pretty simple.

**Comment by palianah** [comment]
Yes sure. 
I Have a commandButton which triggers to show the Form (it is simplified a bit, but the main goal is the same).
```
<p:commandButton id="commandButton"
                                 onclick="$('#calendarForm').css('display', 'block');"
                                 iconPos="right"
                                 value="Show Datepicker" />
```

And I Have a form which wraps the calendar, and this form is displayed to none so the calendar is also not visible on load. 
```
<h:form id="calendarForm" styleClass="calendarForm" style="display:none;">
    <p:calendar value="#{datePanelBean.selectedDate.from}"
                         pagedate="#{datePanelBean.getPageDate()}"
                         readonlyInput="true"
                         styleClass="datepickerClass"
                         id="datePickerId"
                         mode="inline"
                         mindate="#{datePanelBean.selectedDate.minFromDate}"
                         maxdate="#{datePanelBean.selectedDate.maxFromDate}"
                         pages="#{datePanelBean.visibleCalendarMonthsAmount}">
        <p:ajax event="dateSelect" listener="#{datePanelBean.onPickupDateSelect('mq4')}"
                     update="#{updateOnDatePicked}" />
    </p:calendar>
</h:form
```

The error also happens in the primefaces 6.2.27 Version, i think all versions after the initial init of the this.updateDatePickerPosition() is set in a release, i don't know the exact version. 
But the Fix should be very easy to just check if the input is visible, if not just return before doing the updateDatePickerPosition stuff :)

**Comment by melloware** [comment]
Excellent here is a reproducer showing the issue:
[pf-6204.zip](https://github.com/primefaces/primefaces/files/5041305/pf-6204.zip)

**Comment by melloware** [comment]
OK this was fixed for 9.0 by this issue: https://github.com/primefaces/primefaces/issues/5898

**Comment by melloware** [comment]
If you need this fix right now just add this JS to your application JS file.
```javascript
$.datepicker._updateDatePickerPosition = function(inst) {
    if (inst.inline) {
        // GitHub #5898 do not fix inline calendar
        return;
    }
    var input = inst.input[0];
    if (!$.datepicker._pos) { // position below input
        $.datepicker._pos = $.datepicker._findPos(input);
        $.datepicker._pos[1] += inst.input[0].offsetHeight; // add the height
    }

    var offset = {
        left: $.datepicker._pos[0],
        top: $.datepicker._pos[1]
    };
    $.datepicker._pos = null;
    var isFixed = false;
    $(input).parents().each(function() {
        isFixed |= $(this).css("position") === "fixed";
        return !isFixed;
    });
    var checkedOffset = $.datepicker._checkOffset(inst, offset, isFixed);
    inst.dpDiv.css({
        top: checkedOffset.top + "px"
    });
};
```

**Comment by palianah** [comment]
Ok thanks for the suggestion. When is the release date from 9.0?

**Comment by palianah** [comment]
@melloware is it possible to add a minor fix version to 8.0.4 or something like that? 
Just asking because we are planing a update to 8.0.3 and would be a little bit bad if we had to wait for this fix for 9.0. 
We can fix it right now as you wrote, but would be cleaner if it is fixed directly in the source code from primefaces so just asking:)

**Comment by melloware** [comment]
I don't work for PrimeTek so its up to @mertsincan to get merged into Elite 8.0.4.

**Comment by mertsincan** [comment]
Fixed for 8.0.4.

Best Regards,
